modular_splurt/code/datums/components/pregnancy.dm

```diff
@@ -233,18 +221,58 @@
 	if(carrier && (stage == max_stage) && prob(2))
 		to_chat(carrier, span_warning("\The [parent] moves, it's probably ready to hatch!"))
 
-/datum/component/pregnancy/proc/hatch(datum/source, obj/item/I, mob/user, params)
+/datum/component/pregnancy/proc/handle_hatch(datum/source, obj/item/I, mob/user, params)
```

Comment:
I'm guessing you gotta now register this one on COMSIG_PARENT_ATTACKBY rather than .proc/hatch